src/util/mdb-exportjson.c

```diff
@@ -0,0 +1,186 @@
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "mdbtools.h"
+
+#ifdef DMALLOC
+#include "dmalloc.h"
+#endif
+
+#undef MDB_BIND_SIZE
+#define MDB_BIND_SIZE 200000
+
+#define is_quote_type(x) (x==MDB_TEXT || x==MDB_OLE || x==MDB_MEMO || x==MDB_DATETIME || x==MDB_BINARY || x==MDB_REPID)
+#define is_binary_type(x) (x==MDB_OLE || x==MDB_BINARY || x==MDB_REPID)
+
+static char *quote_char = "\"";
+static char *escape_char = "\\";
+static char *separator_char = ":";
+static char *row_start = "{";
+static char *row_end = "}\n";
+static char *delimiter = ",";
+static size_t quote_len = 1; //strlen(quote_char); /* multibyte */
+static size_t orig_escape_len = 1; //strlen(escape_char);
+static int drop_nonascii;
+
+
+//#define DONT_ESCAPE_ESCAPE
+static void
+print_quoted_value(FILE *outfile, char* value, int bin_len) {
+	fputs(quote_char, outfile);
+	int is_binary = (bin_len != -1);
+	while (1) {
+		if (is_binary) {
+			if (!bin_len--)
+				break;
+		} else /* use \0 sentry */
+			if (!*value)
+				break;
+
+		if (quote_len && !strncmp(value, quote_char, quote_len)) {
+			fprintf(outfile, "%s%s", escape_char, quote_char);
+			value += quote_len;
+#ifndef DONT_ESCAPE_ESCAPE
+		} else if (orig_escape_len && !strncmp(value, escape_char, orig_escape_len)) {
+			fprintf(outfile, "%s%s", escape_char, escape_char);
+			value += orig_escape_len;
+#endif
+		} else if (*value < 0x20) {
+			if (!is_binary || drop_nonascii) {
+				putc(' ', outfile);
+				++value;
+			} else {
+				// escape control codes / binary data.
+				fprintf(outfile, "\\x%02x", *(unsigned char*)value++);
+			}
+		} else {
+			putc(*value++, outfile);
+		}
+	}
+	fputs(quote_char, outfile);
+}
+
+static void
+print_col(FILE *outfile, char* col_name, gchar *col_val, int col_type, int bin_len) {
+	print_quoted_value(outfile, col_name, -1);
+	fputs(separator_char, outfile);
+	if (is_quote_type(col_type)) {
+		if (!is_binary_type(col_type)) {
+			bin_len = -1;
+		}
+		print_quoted_value(outfile, col_val, bin_len);
+	} else
+		fputs(col_val, outfile);
+}
+int
+main(int argc, char **argv)
+{
+	unsigned int i;
+	MdbHandle *mdb;
+	MdbTableDef *table;
+	MdbColumn *col;
+	char **bound_values;
+	int  *bound_lens;
+	FILE *outfile = stdout;
+	drop_nonascii = 1;
+	int  opt;
+	char *value;
+	size_t length;
+
+	while ((opt=getopt(argc, argv, "AD:"))!=-1) {
+		switch (opt) {
+		case 'A':
+			drop_nonascii = 0;
+		break;
+		case 'D':
+			mdb_set_date_fmt(optarg);
+		break;
+		default:
+		break;
+		}
+	}
+	/*
+	** optind is now the position of the first non-option arg,
+	** see getopt(3)
+	*/
+	if (argc-optind < 2) {
+		fprintf(stderr,"Usage: %s [options] <file> <table>\n",argv[0]);
+		fprintf(stderr,"where options are:\n");
+		fprintf(stderr,"  -D <format>          set the date format (see strftime(3) for details)\n");
+		fprintf(stderr,"  -A                   drop non ascii characters in non-binary fields\n");
```

Comment:
By default `drop_nonascii=1`. If `-A` is specified, then `drop_nonascii=0`. So `-A` flag actually **disables** dropping of non-ascii characters.